src/MQTTClient.cpp

```diff
@@ -140,12 +140,32 @@ MQTTClient::~MQTTClient() {
   free(this->writeBuf);
 }
 
-void MQTTClient::begin(const char _hostname[], int _port, Client &_client) {
+void MQTTClient::begin(IPAddress ipAddr, int port, Client &client) {
   // set hostname and port
-  this->setHost(_hostname, _port);
+  this->setHost(ipAddr, port);
 
   // set client
-  this->netClient = &_client;
+  this->netClient = &client;
+
+  // initialize client
```

Comment:
The initialization should not be copied and instead extracted in a function that is called by both *begin variants.

Reply:
OK, will do